enumerables.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength,
+module Enumerable
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_each
+    self_item = self
+    if block_given?
+      self_item.size.times do |i|
+        yield(self_item[i])
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    self_item = self
+    if block_given?
+      i = 0
+      while self_item.size > i
+        yield(self_item[i], i)
+        i += 1
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    self_item = self
+    if block_given?
+      rst = []
+      self_item.my_each { |i| rst.push(i) if yield(i) }
+      rst
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(pattern = nil)
+    y = true
+    self_item = self
+    if block_given?
+      self_item.my_each { |x| y = false unless yield(x) }
+    elsif pattern
+      self_item.my_each { |x| y = false unless pattern == x }
+    elsif pattern.is_a? Regexp
+      self_item.my_each { |x| y = false unless x =~ pattern }
+    elsif pattern.is_a? Class
+      self_item.my_each { |x| y = false unless x.is_a? pattern }
+    else
+      self_item.my_each { |x| y = false unless x }
+    end
+    y
+  end
+
+  def my_any?(pattern = nil)
+    y = false
+    self_item = self
+    if block_given?
+      self_item.my_each { |x| y = true if yield(x) }
+    elsif pattern
+      self_item.my_each { |x| y = true if pattern == x }
+    elsif pattern.is_a? Regexp
+      self_item.my_each { |x| y = true if x =~ pattern }
+    elsif pattern.is_a? Class
+      self_item.my_each { |x| y = true if x.is_a? pattern }
+    else
+      self_item.my_each { |x| y = true if x }
+    end
+    y
+  end
+
+  def my_none?(pattern = nil)
+    y = true
+    self_item = self
+    if block_given?
+      self_item.my_each { |x| y = false if yield(x) }
+    elsif pattern
+      self_item.my_each { |x| y = false if pattern == x }
+    elsif pattern.is_a? Regexp
+      self_item.my_each { |x| y = false if x =~ pattern }
+    elsif pattern.is_a? Class
+      self_item.my_each { |x| y = false if x.is_a? pattern }
+    else
+      self_item.my_each { |x| y = false if x }
+    end
+    y
+  end
```

Comment:
#### Failed tests in `my_none?` method
- enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex